assume.go

```diff
@@ -48,24 +48,24 @@ func assumeRole(account, role string) {
 	role = adjustRoleName(role)
 
 	err := tryToAssumeRole(account, role)
-	if err != nil {
-		needToBeRotatedChan := make(chan bool)
-		go isNeedRotateKey(needToBeRotatedChan)
 
+	if err != nil {
 		getMainAccountMfaSessionToken()
 		err = tryToAssumeRole(account, role)
 		if err != nil {
 			log.Fatalln(err)
 		}
 
+		needToBeRotatedChan := make(chan bool)
+		go isNeedRotateKey(needToBeRotatedChan)
```

Comment:
If you start the goroutine and then immediately reads from the channel the goroutine writes into, you can do it directly in the main goroutine. In any case, the user has to wait for this decision independently whether the key will be rotated or not.